### PR TITLE
Implements a simple WiFi AP with promiscuous frame reception

### LINKED ISSUE #12: Implement WiFi Access Point
### Description
Implement dummy WiFi AP functionality for ESP32. It should be able to support up to 4 simultaneous clients, enabling connectivity between them.

### Expected artifacts
WiFi AP source code pushed to repository.

### PR DESCRIPTION
**Pull request by alexfrt**
Closes #12 #15

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -1,17 +1,10 @@
...
 [env:ttgo-lora32-v1]
 platform = espressif32
 board = ttgo-lora32-v1
 framework = arduino
 monitor_speed = 115200
 lib_deps = 
-    adafruit/Adafruit SSD1306@^2.4.2
\ No newline at end of file
+	adafruit/Adafruit SSD1306@^2.4.2
+build_flags =
```
Comment (pxds) [inline platformio.ini:8]: -Wall with -Werror will stump a lot of builds, idk if that's a good idea.

File: lib/network/src/layer2.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#ifndef _LAYER2_CPP_
```
Comment (pxds) [inline lib/network/src/layer2.cpp:1]: why not inline all this stuff on the header file?

File: lib/network/src/layer2.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    layer2data.type = (ether_type_t)ntohs(llc->type);
+    layer2data.payload = payload;
+    for (int i = 0; i < 6; i++) {
+      layer2data.source[i] = mac->addr2[i];
+      layer2data.destination[i] = mac->addr3[i];
+    }
+  }
+
+  return layer2data;
+}
```
Comment (pxds) [inline lib/network/src/layer2.cpp:57]: is this expected to increase in size/complexity? couldn't this all be inlined to the header file for simplicity's sake?

File: lib/wifi/src/ap.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#ifndef _AP_H_
+#define _AP_H_
+
+#include <WiFi.h>
+#include <esp_wifi.h>
```
Comment (pxds) [inline lib/wifi/src/ap.h:5]: why both libs are needed?

File: lib/wifi/src/ap.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      Serial.print(".");
+      Serial.print(ipv4->source[3]);
+      Serial.print(" DST IPv4: ");
+      Serial.print(ipv4->destination[0]);
+      Serial.print(".");
+      Serial.print(ipv4->destination[1]);
+      Serial.print(".");
+      Serial.print(ipv4->destination[2]);
+      Serial.print(".");
+      Serial.println(ipv4->destination[3]);
```
Comment (pxds) [inline lib/wifi/src/ap.cpp:85]: this is a code smell, we don't usually print this much stuff inside class' implementation, return the data and print it on main.cpp or make a dedicated print function.

**Review by pxds — CHANGES_REQUESTED** [review]
just a quick change :)

File: platformio.ini (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexfrt) [reply platformio.ini:8]: I think it's better to have it now in the beginning of the project so in the future we don't need to fix any warnings... But if you think it's cause much more work than we need now, I can remove the -Werror flag

File: lib/network/src/layer2.cpp (hunk 3 above)
Comment (alexfrt) [reply lib/network/src/layer2.cpp]: Nope, it just checks if the frame data is compliant with RFC 1042. It's been extracted to a method. Thanks.

File: platformio.ini (hunk 1 above)
[the remark this answers is not in the document]
Comment (pxds) [reply platformio.ini:8]: lets think about this bridge when we get there.

File: lib/network/src/layer2.cpp [hunk 6]
```diff
@@ -14,10 +16,26 @@ static inline ieee80211_frame_type_t getFrameType(void* buffer) {
   return isQoS ? WIFI_QOSDATA_FRAME : WIFI_DATA_FRAME;
 }
 
-layer2_data_t getLayer2Data(void* buffer) {
+static inline bool isRFC1042Compliant(ieee80211_llc_headers_t* llc) {
```
Comment (pxds) [inline lib/network/src/layer2.cpp:19]: can we have a comment above this function explaining the llc values?

**Review by pxds — APPROVED** [review]

**Pull request merged**